### PR TITLE
refactor: await and promise usage fixes

### PR DESCRIPTION
**Pull request by 0xApotheosis**
## Description

We use promises and awaits in a lot of places that we shouldn't.

This PR puts an end to that by fixing the offending instances and adding a lint rule (`"require-await": "error"`) to stop new offences.

## Notice

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/blob/develop/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [x] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

N/A

## Risk

Small.

## Testing

A general app smoke test should be done.

### Engineering

☝️ 

### Operations

☝️ 

## Screenshots (if applicable)

N/A

### CODE REVIEW

File: src/components/Modals/FiatRamps/config.ts [hunk 1]
```diff
@@ -109,10 +109,10 @@ export const supportedFiatRamps: SupportedFiatRamp = {
     isActive: () => true,
     minimumSellThreshold: 50,
     order: 3,
-    getBuyAndSellList: async () => {
+    getBuyAndSellList: () => {
       const buyAssetIds = adapters.getSupportedBanxaAssets().map(({ assetId }) => assetId)
       const sellAssetIds = [btcAssetId, usdcAssetId, usdtAssetId]
-      return [buyAssetIds, sellAssetIds]
+      return Promise.resolve([buyAssetIds, sellAssetIds])
```
Comment (gomesalexandre) [inline src/components/Modals/FiatRamps/config.ts:115]: 🧠

File: src/components/Modals/Send/hooks/useFormSend/useFormSend.test.tsx [hunk 2]
```diff
@@ -190,7 +190,7 @@ describe.each([
     ;(supportsETH as unknown as jest.Mock<unknown>).mockReturnValue(true)
 
     const sendClose = jest.fn()
-    ;(ensLookup as unknown as jest.Mock<unknown>).mockImplementation(async () => ({
+    ;(ensLookup as unknown as jest.Mock<unknown>).mockImplementation(() => ({
```
Comment (gomesalexandre) [inline src/components/Modals/Send/hooks/useFormSend/useFormSend.test.tsx:193]: `ensLookup` is of type `((domain: string) => Promise<ResolveVanityAddressReturn>) & MemoizedFunction`, I reckon this should return a
```suggestion
    ;(ensLookup as unknown as jest.Mock<unknown>).mockImplementation(() => Promise.resolve({
```

**Review by gomesalexandre — COMMENTED** [review]
Refactor looking very tight ser 🐐

Did a send/receive/fiat ramps/DeFi/portfolio smoke test and didn't find any regression

Added on comment WRT `ensLookup` refactored tests

File: src/components/Modals/Send/hooks/useFormSend/useFormSend.test.tsx (hunk 2 above)
Comment (0xApotheosis) [reply src/components/Modals/Send/hooks/useFormSend/useFormSend.test.tsx]: Checks out!

**Review by 0xdef1cafe — APPROVED** [review]

**Review by gomesalexandre — APPROVED** [review]

**Pull request merged**